command/build_test.go

```diff
@@ -368,29 +368,61 @@ func TestBuildExceptFileCommaFlags(t *testing.T) {
 	c := &BuildCommand{
 		Meta: testMetaFile(t),
 	}
-
-	args := []string{
-		"-parallel-builds=1",
-		"-except=chocolate,vanilla",
-		filepath.Join(testFixture("build-only"), "template.json"),
+	tc := []struct {
+		name                     string
+		args                     []string
+		expectedFiles            []string
+		buildNotExpectedFiles    []string
+		postProcNotExpectedFiles []string
+	}{
+		{
+			name: "JSON: except build and post-processor",
+			args: []string{
+				"-parallel-builds=1",
+				"-except=chocolate,vanilla,pear",
+				filepath.Join(testFixture("build-only"), "template.json"),
+			},
+			expectedFiles:            []string{"apple.txt", "cherry.txt", "peach.txt"},
+			buildNotExpectedFiles:    []string{"chocolate.txt", "vanilla.txt", "tomato.txt", "unnamed.txt"},
+			postProcNotExpectedFiles: []string{"pear.txt"},
+		},
+		{
+			name: "HCL2: except build and post-processor",
+			args: []string{
+				"-parallel-builds=1",
+				"-except=file.chocolate,file.vanilla,pear",
+				filepath.Join(testFixture("build-only"), "template.pkr.hcl"),
+			},
+			expectedFiles:            []string{"apple.txt", "cherry.txt", "peach.txt"},
+			buildNotExpectedFiles:    []string{"chocolate.txt", "vanilla.txt", "tomato.txt", "unnamed.txt"},
+			postProcNotExpectedFiles: []string{"pear.txt"},
+		},
```

Comment:
I just re-read the docs and I think it would be a good idea to also test that `Any post-processor following a skipped post-processor will not run` ? ([docs](https://www.packer.io/docs/commands/build/#except-foo-bar-baz))